lib/pbench/server/api/resources/query_apis/datasets_detail.py

```diff
@@ -24,7 +24,6 @@ def __init__(self, config: PbenchServerConfig, logger: Logger):
             Schema(
                 Parameter("user", ParamType.USER, required=False),
                 Parameter("access", ParamType.ACCESS, required=False),
-                Parameter("name", ParamType.STRING, required=True),
```

Comment:
Although with my new schema changes the `dataset` name will move to the `uri_schema`, right now `_dispatch` merges them, and I don't really like removing it from the schema entirely. (Even though we don't drop unrecognized parameters, so it's not rejected, and Flask will reject the URI mapping if it's not provided.) I'd rather keep it in the schema if only for documentation purposes!

Reply:
Yeah, the problem right now is if keep the `name` schema for the documentation purpose then we also have to keep the `required` parameter as `True`. But then the `test_missing_keys` test fails since we are not actually providing the `name` in the json payload (for something that is required), it is coming from the URI parameterization. Also, `name` parameter validation will always be true in this case because without having a `name` parameter, flask will return 404 on our `dataset/details/` endpoint.

Reply:
Well, we have `uri_parameter` as one of the parameters in the schema that we can set to True. So Nevermind :smiley:

Reply:
Yeah; and with my schema change (TBD whether you merge or I do 😆 ) the uri parameters move to their own schema object so it'll be even more clear. (At some point I'd love to rewrite `Common` to take better advantage of all this.)